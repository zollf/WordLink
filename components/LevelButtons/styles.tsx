import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  button: {
    marginBottom: 24,
  },
});

export default styles;