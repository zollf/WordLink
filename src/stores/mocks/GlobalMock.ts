const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
