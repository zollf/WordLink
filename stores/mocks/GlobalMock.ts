const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
