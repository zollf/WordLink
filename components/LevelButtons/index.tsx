import React, { useEffect } from 'react';
import { View, Animated } from "react-native";
import { useStore, useAnimate } from '../../hooks';
import Button from '../Button';

import { ElasticSlideIn, FadeSlideIn } from '../../animations';
import styles from './styles';

const LevelButtons = () => {
  const { setCurrentDifficultyOpen } = useStore('global');

  const [easy, easy_opacity] = useAnimate(-100, 0);
  const [medium, medium_opacity] = useAnimate(-100, 0);
  const [hard, hard_opacity] = useAnimate(-100, 0);

  useEffect(() => {
    Animated.parallel([
      Animated.stagger(50, [
        ElasticSlideIn(easy),
        ElasticSlideIn(medium),
        ElasticSlideIn(hard)
      ]),
      Animated.stagger(50, [
        FadeSlideIn(easy_opacity),
        FadeSlideIn(medium_opacity),
        FadeSlideIn(hard_opacity),
      ]),
    ]).start();
  });

  return (
    <View>
      <Animated.View style={[styles.button, {
        transform: [{translateX: easy}],
        opacity: easy_opacity,
      }]}>
        <Button text="EASY" color="green" onPress={() => setCurrentDifficultyOpen('easy')}/>
      </Animated.View>

      <Animated.View style={[styles.button, {
        transform: [{translateX: medium}],
        opacity: medium_opacity,
      }]}>
        <Button text="MEDIUM" color="orange_2" onPress={() => setCurrentDifficultyOpen('medium')}/>
      </Animated.View>

      <Animated.View style={[styles.button, {
        transform: [{translateX: hard}],
        opacity: hard_opacity,
      }]}>
        <Button text="HARD" color="red" onPress={() => setCurrentDifficultyOpen('hard')}/>
      </Animated.View>
    </View>
  );
};

export default LevelButtons;