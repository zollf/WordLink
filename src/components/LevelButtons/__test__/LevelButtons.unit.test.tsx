import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from '../../../stores/mocks/GlobalMock';

import LevelButtons from '..';

import renderer from 'react-test-renderer';

const component = (
  <Provider global={GlobalMock}>
    <LevelButtons />
  </Provider>
);

describe('<MenuBar/>', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
