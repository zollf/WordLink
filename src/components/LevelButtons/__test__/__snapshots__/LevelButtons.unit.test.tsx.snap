// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBar/> matches snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "opacity": mockConstructor {},
          "transform": Array [
            Object {
              "translateX": mockConstructor {},
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": mockConstructor {},
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4282711882,
                  4280928549,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "height": 106.5,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Raleway_700Bold",
                  "fontSize": 33.5,
                }
              }
            >
              EASY
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "opacity": mockConstructor {},
          "transform": Array [
            Object {
              "translateX": mockConstructor {},
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": mockConstructor {},
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294956160,
                  4294950464,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "height": 106.5,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Raleway_700Bold",
                  "fontSize": 33.5,
                }
              }
            >
              MEDIUM
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "opacity": mockConstructor {},
          "transform": Array [
            Object {
              "translateX": mockConstructor {},
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": mockConstructor {},
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294918208,
                  4290451737,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "height": 106.5,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Raleway_700Bold",
                  "fontSize": 33.5,
                }
              }
            >
              HARD
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;
